src/streamsync/core.py

```diff
@@ -723,7 +724,8 @@ class Evaluator:
 
     template_regex = re.compile(r"[\\]?@{([\w\s.]*)}")
 
-    def __init__(self, session_state: StreamsyncState):
+    def __init__(self, session_id: str, session_state: StreamsyncState):
+        self.session_id = session_id
```

Comment:
Why are we passing an id instead of following what was done with `session_state` and passing component like `session_components`?

Reply:
The id is currently needed to access component manager inside session(s) – previously, Evaluator used "application-wide" ComponentManager in the `evaluate_field` method; to replace it with session-linked manager, we first need to know what session we're working on specifically. I might've misunderstood the purpose of the StreamsyncState, but I couldn't see a way to utilize it for this.

Reply:
But what I mean is, follow exactly what has been done with State. Rather than getting the session by id and then getting the component manager, pass the reference to the component manager when instantiating the Evaluator, similarly to how we pass `session_state`.

Reply:
Ah, I see, I've misunderstood your comment, apologies. 

Good point, as session context is accessible during the creation of Evaluator, we might as well just pass the manager, without having the overhead of resolving it each time. I'll adjust it.